router/router.go

```diff
@@ -1993,6 +2020,7 @@ func (rt *HandleT) readAndProcess() int {
 
 			rt.timeGained += latenciesUsed[job.WorkspaceId]
 
+			rt.responseQ <- jobResponseT{status: &status, worker: nil, userID: job.UserID, JobT: job}
```

Comment:
why don't we simply update the reporting stats here instead of passing drained jobs to the `responseQ` channel and having to deal with all the associated (extra) complexities?

I don't believe that at this point we care too much about some extra reporting code duplication, as long as we keep the rest of the design cleaner. We should be revisiting the whole reporting design sometime soon anyway!

Reply:
My initial implementation did involve copying over the relevant parts of the code from the commitStatusList function to manually commit the information to the report. However, the code duplication ended up being more that I felt comfortable with. I will push another commit with that implementation for you guys to review and take a call between the 2 implementation.